src/libs/Network/SequentialQueue.ts

```diff
@@ -128,7 +128,7 @@ function process(): Promise<void> {
     return currentRequestPromise;
 }
 
-function flush() {
+function flush(shouldResetPromise = true) {
```

Comment:
NAB: Can you please add a doc comment explaining what this param does and why it's necessary. You could basically describe the RCA of the issue.